entries/block.go

```diff
@@ -34,12 +35,43 @@ type PGBlockEntry struct {
 	BlockEntry
 }
 
+type BlockSigner struct {
+	BlockHash   string
+	SignerIndex uint64
+}
+
+type PGBlockSigner struct {
+	bun.BaseModel `bun:"table:block_signer"`
+	BlockSigner
+}
+
 // Convert the UserAssociation DeSo encoder to the PG struct used by bun.
-func BlockEncoderToPGStruct(block *lib.MsgDeSoBlock, keyBytes []byte, params *lib.DeSoParams) *PGBlockEntry {
+func BlockEncoderToPGStruct(block *lib.MsgDeSoBlock, keyBytes []byte, params *lib.DeSoParams) (*PGBlockEntry, []*PGBlockSigner) {
 	blockHash, _ := block.Hash()
+	blockHashHex := hex.EncodeToString(blockHash[:])
+	qc := block.Header.GetQC()
+	blockSigners := []*PGBlockSigner{}
+	if !isInterfaceNil(qc) {
```

Comment:
Is this safe? Please make this this works for a PoW block where the header has neither a QC nor an aggregate QC

Reply:
yeah this is safe. the `isInterfaceNil` check returns true for pow blocks.